src/redux/books/books.js

```diff
@@ -1,9 +1,35 @@
-import React from 'react';
+const ADD_BOOK = 'redux/books/ADD_BOOK';
+const REMOVE_BOOK = 'redux/books/REMOVE_BOOK';
 
-const BooksComponent = () => (
-  <div>
-    <h1>Books</h1>
-  </div>
-);
+const initialState = [];
 
-export default BooksComponent;
+const addBook = (c, t, a) => ({
+  type: ADD_BOOK,
+  category: c,
+  title: t,
+  author: a,
+});
+
+const removeBook = (i) => ({
+  type: REMOVE_BOOK,
+  id: i,
+});
+
+const booksReducer = (state = initialState, action) => {
+  switch (action.type) {
+    case ADD_BOOK:
+      state.push({
+        id: state.length + 1,
+        category: action.category,
+        title: action.title,
+        author: action.author,
+      });
```

Comment:
- [ ] This is an anti-pattern as the state shouldn't be updated. You can simply create a new array and push the new content with the previous state there then return that array. Kindly fix it.